File: pages/login_page.py
from selenium.webdriver.common.by import By
from pages.base_page import BasePage

class LoginPage(BasePage):
    # Locators
    EMAIL_INPUT = (By.ID, 'email')
    PASSWORD_INPUT = (By.ID, 'password')
    LOGIN_BUTTON = (By.XPATH, '//button[@type='submit']')
    ERROR_MESSAGE = (By.CLASS_NAME, 'error-message')
    FORGOT_PASSWORD_LINK = (By.LINK_TEXT, 'Forgot Password?')
    SIGNUP_LINK = (By.LINK_TEXT, 'Sign Up')

    def __init__(self, driver):
        super().__init__(driver)

    def enter_email(self, email):
        self.enter_text(self.EMAIL_INPUT, email)

    def enter_password(self, password):
        self.enter_text(self.PASSWORD_INPUT, password)

    def click_login_button(self):
        self.click(self.LOGIN_BUTTON)

    def login(self, email, password):
        self.enter_email(email)
        self.enter_password(password)
        self.click_login_button()

    def get_error_message(self):
        return self.get_text(self.ERROR_MESSAGE)

    def is_error_message_present(self):
        return self.is_element_present(self.ERROR_MESSAGE)

    def click_forgot_password(self):
        self.click(self.FORGOT_PASSWORD_LINK)

    def click_signup(self):
        self.click(self.SIGNUP_LINK)

File: pages/loginPage.js
// LoginPage class for Gypsy Nurse application
const BasePage = require('./basePage');

class LoginPage extends BasePage {
  constructor(page) {
    super(page);
    // Locators
    this.EMAIL_INPUT = '#email';
    this.PASSWORD_INPUT = '#password';
    this.LOGIN_BUTTON = 'button[type="submit"]';
    this.ERROR_MESSAGE = '.error-message';
    this.FORGOT_PASSWORD_LINK = 'text=Forgot Password?';
    this.SIGNUP_LINK = 'text=Sign Up';
  }

  async enterEmail(email) {
    await this.enterText(this.EMAIL_INPUT, email);
  }

  async enterPassword(password) {
    await this.enterText(this.PASSWORD_INPUT, password);
  }

  async clickLoginButton() {
    await this.click(this.LOGIN_BUTTON);
  }

  async login(email, password) {
    await this.enterEmail(email);
    await this.enterPassword(password);
    await this.clickLoginButton();
  }

  async getErrorMessage() {
    return await this.getText(this.ERROR_MESSAGE);
 }

  async isErrorMessagePresent() {
    return await this.isElementPresent(this.ERROR_MESSAGE);
  }

  async clickForgotPassword() {
    await this.click(this.FORGOT_PASSWORD_LINK);
  }

  async clickSignup() {
    await this.click(this.SIGNUP_LINK);
  }
}

module.exports = LoginPage;
